api/api.go

```diff
@@ -175,14 +177,16 @@ func NewAPIWithVersion(ctx context.Context, globalConfig *conf.GlobalConfigurati
 			r.Post("/generate_link", api.GenerateLink)
 		})
 
-		r.Route("/saml", func(r *router) {
-			r.Route("/acs", func(r *router) {
-				r.Use(api.loadSAMLState)
-				r.Post("/", api.ExternalProviderCallback)
-			})
+		if api.config.SAML.Enabled {
```

Comment:
I would consider putting this into a middleware function rather than hiding the endpoint if it's not enabled.

If it's not enabled and someone hits the SAML endpoint, we can return an error like "Contact your admin to enable SAML" which would be more explicit.

Reply:
Kinda like how we don't hide the /authorize and /callback endpoints if no oauth2 providers are enabled

Reply:
I'll take care of this in another PR, if that's all right?